drivers/npmx/npmx_shell.c

```diff
@@ -604,9 +604,9 @@ static int cmd_buck_vout_select_set(const struct shell *shell, size_t argc, char
 	}
 
 	int err = 0;
-	uint8_t buck_indx = (uint8_t)shell_strtoul(argv[1], 0, &err);
+	uint8_t buck_indx = CLAMP(shell_strtoul(argv[1], 0, &err), 0, UINT8_MAX);
 	npmx_buck_vout_select_t selection =
-		(npmx_buck_vout_select_t)shell_strtoul(argv[2], 0, &err);
+		(npmx_buck_vout_select_t)CLAMP(shell_strtoul(argv[2], 0, &err), 0, UINT8_MAX);
```

Comment:
I think this one could be clamped to 2 or max value of `npmx_buck_vout_select_t` if it exists. Then checking value of `selection` in lines 621-625 could be removed.

Reply:
We don't have max value for `npmx_buck_vout_select_t`, this is good idea as long as those values are `0` and `1` (right now they are, but those registers likes to change).

Reply:
Nope, it can't be clamped to the max value - in this case every argument will be treated as valid, because it will be clamped to the max  value, which is valid. Ofc we can clamp to max+1, but in this case  the "checking value of selection in lines 621-625" will transform into conditional statement that checks if the passed value is different than max+1, so there will be no improvement.